### PR TITLE
Better command API

### PR DESCRIPTION
**Pull request by ramidzkh**
Use Brigadier

### CODE REVIEW

**Review by hYdos — APPROVED** [review]
epic

File: build.gradle [hunk 1]
```diff
@@ -87,6 +87,8 @@ allprojects {
 		modCompile("net.fabricmc:fabric-loader-1.8.9:0.8.2+build.202004131640") {
 			exclude module: "guava"
 		}
+
+		implementation 'com.mojang:brigadier:1.0.17'
```
Comment (BoogieMonster1O1) [inline build.gradle:91]: put this in the module's build.gradle 
i don't think any other module will use it

File: fabric-commands-v1/src/main/java/net/fabricmc/fabric/api/command/v1/DispatcherRegistrationCallback.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+			DispatcherRegistrationCallback.class,
+			(dispatcher, isDedicatedServer) -> {
+			},
+			callbacks -> (dispatcher, isDedicatedServer) -> {
+				for (DispatcherRegistrationCallback callback : callbacks) {
+					callback.initialize(dispatcher, isDedicatedServer);
+				}
+			});
+
+	void initialize(CommandDispatcher<ServerCommandSource> dispatcher, boolean isDedicatedServer);
```
Comment (BoogieMonster1O1) [inline fabric-commands-v1/src/main/java/net/fabricmc/fabric/api/command/v1/DispatcherRegistrationCallback.java:35]: `isDedicatedServer` -> `dedicated`

File: fabric-commands-v1/src/main/java/net/fabricmc/fabric/mixin/command/CommandManagerMixin.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.server.command.CommandManager;
+
+import net.fabricmc.fabric.api.command.v1.DispatcherRegistrationCallback;
+import net.fabricmc.fabric.impl.command.CommandManagerHolder;
+
+@Mixin(CommandManager.class)
+public class CommandManagerMixin {
```
Comment (BoogieMonster1O1) [inline fabric-commands-v1/src/main/java/net/fabricmc/fabric/mixin/command/CommandManagerMixin.java:32]: did we start using suffixes now?

File: fabric-commands-v1/src/main/java/net/fabricmc/fabric/mixin/command/CommandRegistryMixin.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+import net.minecraft.server.command.CommandRegistry;
+import net.minecraft.text.LiteralText;
+
+import net.fabricmc.fabric.api.command.v1.ServerCommandSource;
+import net.fabricmc.fabric.impl.command.CommandManagerHolder;
+
+@Mixin(CommandRegistry.class)
+public class CommandRegistryMixin {
+	@Inject(method = "execute", at = @At("HEAD"), cancellable = true)
+	private void execute(CommandSource source, String command, CallbackInfoReturnable<Integer> callbackInfoReturnable) {
```
Comment (BoogieMonster1O1) [inline fabric-commands-v1/src/main/java/net/fabricmc/fabric/mixin/command/CommandRegistryMixin.java:39]: `callbackInfoReturnable` -> `info`

**Review by BoogieMonster1O1 — CHANGES_REQUESTED** [review]
e🅱️ic
also make a testmod when it's ready

File: build.gradle (hunk 1 above)
Comment (ramidzkh) [reply build.gradle:91]: breaks with the run config since running only uses the classpath of the root project

**Comment by hYdos** [comment]
i approve. shedaniel wont approve i merge

**Comment by BoogieMonster1O1** [comment]
what just happened

**Pull request merged**